modules/swagger-jaxrs/src/main/java/com/wordnik/swagger/jaxrs/ParameterProcessor.java

```diff
@@ -73,26 +76,37 @@ public static Parameter applyAnnotations(Swagger swagger, Parameter parameter, C
       }
 
       final String defaultValue = param.defaultValue();
-      if (param.allowMultiple() || isArray) {
+      if (p.getItems() != null || param.allowMultiple()) {
+        if (p.getItems() == null) {
+          // Convert to array
+          final Map<PropertyBuilder.PropertyId, Object> args = new EnumMap<PropertyBuilder.PropertyId, Object>(PropertyBuilder.PropertyId.class);
+          args.put(PropertyBuilder.PropertyId.DEFAULT, p.getDefaultValue());
+          p.setDefaultValue(null);
+          args.put(PropertyBuilder.PropertyId.ENUM, p.getEnum());
+          p.setEnum(null);
+          args.put(PropertyBuilder.PropertyId.MINIMUM, p.getMinimum());
+          p.setMinimum(null);
+          args.put(PropertyBuilder.PropertyId.EXCLUSIVE_MINIMUM, p.isExclusiveMinimum());
+          p.setExclusiveMinimum(null);
+          args.put(PropertyBuilder.PropertyId.MAXIMUM, p.getMaximum());
+          p.setMaximum(null);
+          args.put(PropertyBuilder.PropertyId.EXCLUSIVE_MAXIMUM, p.isExclusiveMaximum());
+          p.setExclusiveMaximum(null);
+          Property items = PropertyBuilder.build(p.getType(), p.getFormat(), args);
+          p.type(ArrayProperty.TYPE).format(null).items(items);
+        }
         final Map<PropertyBuilder.PropertyId, Object> args = new EnumMap<PropertyBuilder.PropertyId, Object>(PropertyBuilder.PropertyId.class);
         if (!defaultValue.isEmpty()) {
           args.put(PropertyBuilder.PropertyId.DEFAULT, defaultValue);
         }
-        if (allowableValues != null) {
-          processAllowedValues(allowableValues, true, args);
-        }
-
-        p.items(PropertyBuilder.build(p.getType(), p.getFormat(), args))
-          .type(ArrayProperty.TYPE)
-          .format(null)
-          .collectionFormat("multi");
+        processAllowedValues(allowableValues, true, args);
+        PropertyBuilder.merge(p.getItems(), args);
```

Comment:
StringProperty may contain enum values, so let's merge additional arguments instead of rebuilding items from scratch.